### PR TITLE
Add new keccak hints + update import paths 

### PR DESCRIPTION
**Pull request by fmoletta**
Solves #881 
Depends on #873 
- Add new hints in builtin_keccak/keccak.cairo
- Update import paths for hints, cairo programs & constants

### CODE REVIEW

File: src/hint_processor/builtin_hint_processor/keccak_utils.rs [hunk 1]
```diff
@@ -181,3 +186,199 @@ fn left_pad(bytes_vector: &mut [u8], n_zeros: usize) -> Vec<u8> {
...
+// ids.output1_high, ids.output1_mid = divmod(tmp, 2 ** 128)
+pub fn split_output_mid_low_high(
+    vm: &mut VirtualMachine,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    let binding = get_integer_from_var_name("output1", vm, ids_data, ap_tracking)?;
+    let output1 = binding.as_ref();
+    let (tmp, output1_low) = output1.div_rem(&Felt::from(256_u64.pow(7)));
+    let (output1_high, output1_mid) = tmp.div_rem(&Felt::one().shl(128_u32));
```
Comment (Oppen) [inline src/hint_processor/builtin_hint_processor/keccak_utils.rs:268]:
```suggestion
    let output1_low = output1 & Felt::from((1u64 << (8 * 7)) - 1u64);
    let tmp = output1 >> (8 * 7);
    let output1_high = tmp >> 128;
    let output1_mid = tmp & Felt::from(u128::MAX);
```

**Review by Oppen — APPROVED** [review]
Only minor comments. LGTM.

File: src/hint_processor/builtin_hint_processor/keccak_utils.rs [hunk 2]
```diff
@@ -181,3 +186,199 @@ fn left_pad(bytes_vector: &mut [u8], n_zeros: usize) -> Vec<u8> {
...
+    vm: &mut VirtualMachine,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    input_key: usize,
+    exponent: u32,
+) -> Result<(), HintError> {
+    let inputs_ptr = get_ptr_from_var_name("inputs", vm, ids_data, ap_tracking)?;
+    let binding = vm.get_integer((inputs_ptr + input_key)?)?;
+    let input = binding.as_ref();
+    let (high, low) = input.div_rem(&Felt::from(256_u64.pow(exponent)));
```
[the remark this answers is not in the document]
Comment (fmoletta) [reply src/hint_processor/builtin_hint_processor/keccak_utils.rs]: Added suggestion

**Comment by Oppen** [comment]
About failing pipelines:
- rust.yml is just missing a borrow in one of the changes I suggested before.
- The benchmark ones fail to compile the Cairo programs, I think it has to do with a typing change in cairo-lang.

File: src/hint_processor/builtin_hint_processor/keccak_utils.rs [hunk 3]
```diff
@@ -181,3 +186,199 @@ fn left_pad(bytes_vector: &mut [u8], n_zeros: usize) -> Vec<u8> {
...
+}
+
+// Implements hint: ids.n_words_to_copy, ids.n_bytes_left = divmod(ids.n_bytes, ids.BYTES_IN_WORD)
+pub fn split_n_bytes(
+    vm: &mut VirtualMachine,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    constants: &HashMap<String, Felt>,
+) -> Result<(), HintError> {
+    let n_bytes = get_integer_from_var_name("n_bytes", vm, ids_data, ap_tracking)?;
```
[the remark this answers is not in the document]
Comment (fmoletta) [reply src/hint_processor/builtin_hint_processor/keccak_utils.rs]: Added conversion to u64 (292e3fe)

**Review by juanbono — APPROVED** [review]

**Pull request merged**